component/src/lib.rs

```diff
@@ -0,0 +1,127 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use anyhow::anyhow;
+use async_trait::async_trait;
+use tokio::sync::mpsc::{channel, Receiver, Sender};
+use tokio::sync::oneshot::{
+    channel as oneshotChannel, Receiver as oneshotReceiver, Sender as oneshotSender,
+};
+
+pub type IrrecoverableError = anyhow::Error;
+type JoinHandle = tokio::task::JoinHandle<()>;
+
+/// A Supervisor is instantiated to supervise a task that should always be running.
+/// A running supervisor will start a component task, and ensure that it is restarted
+/// if it ever stops.
+pub struct Supervisor<M: Manageable> {
+    irrecoverable_signal: Receiver<IrrecoverableError>,
```

Comment:
Some docs on each field (even though they are not public) help us remember what they are for when we maintain.

Reply:
👍